src/components/statevector_component/aggregation.py

```diff
@@ -480,11 +489,26 @@ def force_native_res_pixels(config, clusters, sensitivities):
     elif config["Res"] == "0.5x0.625":
         lat_step = 0.5
         lon_step = 0.625
+    
+    for lat, lon in coords:
+        lon = np.floor(lon / lon_step) * lon_step
+        lat = np.floor(lat / lat_step) * lat_step
+
+ 
+    print(f"Length of the list before: {len(coords)}")
+    coords = sorted(set(map(tuple, coords)), reverse=True) # Remove any duplicate coordinates within the same gridcell. 
```

Comment:
You may consider using [numpy.unique](https://numpy.org/doc/stable/reference/generated/numpy.unique.html) for this.

What does map do here?

Reply:
Since the set() function cannot be applied to mutable objects (e.g. lists), I converted the coordinates into tuples using the map() function.